index.js

```diff
@@ -1,253 +1,38 @@
 
-var fs = require('fs'),
-    levelup = require('levelup'),
-    multilevel = require('level-multiply'),
-    parser = require('osm-pbf-parser')(),
-    through = require('through2'),
-    suggester = require('pelias-suggester-pipeline'),
-    propStream = require('prop-stream'),
-    settings = require('pelias-config').generate(),
-    dbclient = require('pelias-dbclient')(),
-    OsmiumStream = require('osmium-stream');
-
-var Backend = require('geopipes-elasticsearch-backend');
-
-// esclient.livestats();
+/**
+  The openstreetmap importer provides a way of parsing, mapping and
+  augmenting OSM data in to elasticsearch.
+**/
 
-var backend = function( index, type ){
-  return new Backend( dbclient.client, index, type );
+var elasticsearch = require('pelias-dbclient'),
+    adminLookup = require('pelias-admin-lookup'),
+    suggester = require('pelias-suggester-pipeline'),
+    dbmapper = require('./stream/dbmapper');
+
+var osm = { pbf: {}, doc: {}, address: {}, tag: {} };
+
+osm.pbf.parser = require('./stream/pbf').parser;
+osm.doc.constructor = require('./stream/document_constructor');
+osm.doc.denormalizer = require('./stream/denormalizer');
+osm.tag.mapper = require('./stream/tag_mapper');
+osm.address.extractor = require('./stream/address_extractor');
+
+// default import pipeline
+osm.import = function(opts){
+  osm.pbf.parser(opts)
+    .pipe( osm.doc.constructor() )
+    .pipe( osm.tag.mapper() )
+    .pipe( osm.doc.denormalizer() )
+    .pipe( adminLookup.stream() )
+    .pipe( osm.address.extractor() )
+    .pipe( suggester.pipeline )
+    .pipe( dbmapper() )
+    .pipe( elasticsearch() );
 };
 
-// @todo: extract this or refactor suggester to make it a stream factory
-var bun = require('bun');
-function generateSuggester(){
-  return bun([
-    suggester.streams.suggestable(),
-    suggester.streams.suggester( suggester.generators )
-  ]);
+// run import if executed directly; but not if imported via require()
+if( require.main === module ){
```

Comment:
This is a fair comment but unfortunately it will need to stay this way until we can update our dependants in `pelias/vagrant` and our build scripts as they currently rely on executing the import via `node index.js`.